threeML/utils/spectrum/spectrum_likelihood.py

```diff
@@ -0,0 +1,315 @@
+from threeML.utils.statistics.likelihood_functions import half_chi2
+from threeML.utils.statistics.likelihood_functions import poisson_log_likelihood_ideal_bkg
+from threeML.utils.statistics.likelihood_functions import poisson_observed_gaussian_background
+from threeML.utils.statistics.likelihood_functions import poisson_observed_poisson_background
+from threeML.exceptions.custom_exceptions import custom_warnings, NegativeBackground
+
+import copy
+import numpy as np
+
+
+
+
+# These classes provide likelihood evaluation to SpectrumLike and children
+
+
+
+class SpectrumLikelihood(object):
+
+    def __init__(self, spectrum_plugin):
+        """
+        
+        A class to hold the likelihood call and randomization of spectrum counts
+        
+        :param spectrum_plugin: the spectrum plugin to call
+        """
+
+        self._spectrum_plugin = spectrum_plugin
+
+
+    def get_log_likelihood(self):
```

Comment:
`get_statistic` or even `get_current_value` might be more appropriate, otherwise, this method has a name very similar to what is used in the plugins to get the likelihood value.

Reply:
👍